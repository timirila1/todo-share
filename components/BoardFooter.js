import React from 'react';
import { StyleSheet, View, AlertIOS } from 'react-native';

import { Footer } from './Footer';
import { LabeledIcon } from './LabeledIcon';
import { deleteBoard } from '../services/Boards';

export const BoardFooter = props => {
    const { navigation, boardId, boardTitle } = props;
    const goToAddTodo = () => navigation.navigate('AddTodo', { boardId });
    const goToEditBoard = () => navigation.navigate('EditBoard', { boardId })
    const deleteCurrentBoard = async () => {
        await deleteBoard(boardId);
        navigation.navigate('Boards');
    };
    const showDeleteAlert = () => {
        AlertIOS.alert(
            `Delete board ${boardTitle}`,
            'Are you sure you want delete this board?',
            [
                {
                    text: 'Cancel',
                    onPress: () => { },
                    style: 'cancel',
                },
                {
                    text: 'Yes, delete',
                    onPress: deleteCurrentBoard,
                },
            ]
        );
    };

    return (
        <Footer>
            <View style={styles.iconWrapper}>
                <LabeledIcon
                    label="Delete"
                    name="trash"
                    onPress={showDeleteAlert}
                />
                <LabeledIcon
                    label="Edit"
                    name="pencil"
                    onPress={goToEditBoard}
                />
                <LabeledIcon
                    label="Share"
                    name="share-apple"
                    onPress={() => { }}
                />
                <LabeledIcon
                    label="Add todo"
                    name="plus"
                    onPress={goToAddTodo}
                />
            </View>
        </Footer>
    );
};

const styles = StyleSheet.create({
    iconWrapper: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-evenly',
    }
});
