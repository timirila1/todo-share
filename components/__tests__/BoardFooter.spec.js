import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { BoardFooter } from '../BoardFooter';

it('Render BoardFooter', async () => {
    const tree = renderer.create(
        <BoardFooter
            navigation={{}}
            boardId={1}
        />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
