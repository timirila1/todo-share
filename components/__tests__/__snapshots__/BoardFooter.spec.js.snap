// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BoardFooter 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderTopColor": "rgba(0, 0, 0, 0.1)",
      "borderTopWidth": 1,
      "bottom": 0,
      "flex": 1,
      "height": 84,
      "left": 0,
      "paddingBottom": 20,
      "position": "absolute",
      "right": 0,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          false,
        ]
      }
    >
      <View
        style={null}
      >
        <Text />
        <Text
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.6)",
                "fontSize": 12,
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Add todo
        </Text>
      </View>
    </View>
  </View>
</View>
`;
